src/commands/createFunction/scriptSteps/ScriptFunctionCreateStep.ts

```diff
@@ -56,8 +62,34 @@ export class ScriptFunctionCreateStep extends FunctionCreateStepBase<IScriptFunc
 
         const language: ProjectLanguage = nonNullProp(context, 'language');
         const fileName: string | undefined = getScriptFileNameFromLanguage(language);
+        await this._installDurableDependenciesIfNeeded(context, language);
+
         return fileName ? path.join(functionPath, fileName) : functionJsonPath;
     }
 
+    private async _installDurableDependenciesIfNeeded(context: IScriptFunctionWizardContext, language: ProjectLanguage): Promise<void> {
+        if (!context.newDurableStorageType) {
+            return;
+        }
+
+        const dfDepInstallFailed: string = localize('failedToAddDurableDependency', 'Failed to add or install durable package dependency. Please inspect and verify if it needs to be added manually.');
+
+        try {
+            switch (language) {
+                case ProjectLanguage.JavaScript:
+                case ProjectLanguage.TypeScript:
+                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'npm', 'install', durableUtils.nodeDfPackage);
+                    break;
+                case ProjectLanguage.Python:
+                    await pythonUtils.addDependencyToRequirements(durableUtils.pythonDfPackage);
+                    await venvUtils.runPipInstallCommandIfPossible();
+                    break;
+                default:
```

Comment:
Add the case `ProjectLanguage.Powershell` with a comment about having to wait for the blocker.